### PR TITLE
The Long Awaited Bootstrap 3 Upgrade

### PR DESCRIPTION
**Pull request by logic-dev**
Solves issue #21 

I've upgraded the site to Bootstrap 3, Font Awesome 4 and angular-strap

This is a huge pull request. I've tried to catch as many bugs as I can but this needs to be tested in staging before we push this out

### CODE REVIEW

File: resources/assets/scripts/app/directives/track-editor.coffee [hunk 1]
```diff
@@ -116,14 +116,14 @@ module.exports = angular.module('ponyfm').directive 'pfmTrackEditor', () ->
...
-                    .then -> $scope.$apply ->
-                        $scope.$emit 'track-deleted'
+                #$dialog.messageBox('Delete ' + $scope.track.title, 'Are you sure you want to delete "' + $scope.track.title + '"?', [
+                #    {result: 'ok', label: 'Yes', cssClass: 'btn-danger'},
+                #    {result: 'cancel', label: 'No', cssClass: 'btn-primary'}
+                #]).open().then (res) ->
+                #    return if res == 'cancel'
+                #    $.post('/api/web/tracks/delete/' + $scope.track.id)
+                #    .then -> $scope.$apply ->
+                #        $scope.$emit 'track-deleted'
```
Comment (Feld0) [inline resources/assets/scripts/app/directives/track-editor.coffee]: Why is all of this commented out?

Comment (logic-dev) [inline resources/assets/scripts/app/directives/track-editor.coffee]: Ah, thought something slipped through the cracks

**Comment by Feld0** [comment]
Can you bring in `angular-strap.js` via npm instead of committing it in?

**Comment by Feld0** [comment]
Can Bootstrap 3 be brought in via npm?

**Comment by logic-dev** [comment]
Switching to npm now

**Comment by Feld0** [comment]
npm'ing Font Awesome would be great, too, for that matter.

**Comment by logic-dev** [comment]
npming all the things!

File: resources/assets/styles/forms.less [hunk 2]
```diff
@@ -36,9 +36,29 @@
...
+  }
+
+  .row .col-sm-12 {
+    padding: 0;
+  }
+
+  .row .col-sm-6 {
+    padding: 0;
+    width: 48.93617021%;
+    margin-left: 2.12765957%;
```
Comment (Feld0) [inline resources/assets/styles/forms.less]: These are some **incredibly** specific percentages - what are they for?

File: resources/assets/styles/layout.less [hunk 3]
```diff
@@ -58,7 +60,7 @@ header {
     }
 
     .dropdown-menu {
-      left: auto;
+      left: auto !important;
```
Comment (Feld0) [inline resources/assets/styles/layout.less]: Can we do without this `!important` statement?

File: resources/assets/styles/forms.less (hunk 2 above)
Comment (logic-dev) [inline resources/assets/styles/forms.less]: I need to find out where they came from I copied them from somewhere in the app using dev tools. I'll try to get the formula instead

File: resources/assets/styles/layout.less (hunk 3 above)
Comment (logic-dev) [inline resources/assets/styles/layout.less]: angular-strap adds style on the element which overwrites left so I need the !important unless there's a toggle in angular-strap. Never worked with that library before :/

Comment (Feld0) [inline resources/assets/styles/layout.less]: All right, then.

**Comment by Feld0** [comment]
Looks good! Let's merge this. :horse:

**Pull request merged**